server/src/session.ts

```diff
@@ -229,9 +229,12 @@ export class Session {
       return;
     }
     for (const change of contentChanges) {
-      if (change.range) {
+      if ('range' in change) {
         const [start, end] = lspRangeToTsPositions(scriptInfo, change.range);
         scriptInfo.editContent(start, end, change.text);
+      } else {
+        // New text is considered to be the full content of the document.
+        scriptInfo.editContent(0, change.text.length, change.text);
```

Comment:
I think the more appropriate method to call in this case is 
```ts
scriptInfo.open(change.text);
```
This is because we need to replace the entire script snapshot with the new content, and not just a portion of the existing snapshot with the new text.
If we were to stick with `scriptInfo.editContent`, the right way to do this should be
```ts
scriptInfo.editContent(0, scriptInfo.getSnapshot().getLength(), change.text);
```
i.e. we want to replace the *entire* snapshot with the new text.

Reply:
You're right. Thanks!

Reply:
It looks like there wasn't any change, did you push your changes?